online-store/src/app/app.ts

```diff
@@ -0,0 +1,278 @@
+
+    checkboxInput.forEach((el) => {
+        if (filtersObj.country) {
+            if (filtersObj.country.includes(el.value)){
+                el.checked = true;
+            }
+        }
+        if (filtersObj.brand) {
+            if (filtersObj.brand.includes(el.value)){
+                el.checked = true;
+            }
+        }               
+    });
+    
+    const sortInput: NodeListOf<HTMLOptionElement> = document.querySelectorAll(".select-input");
+    sortInput.forEach((el) => {
+        if (filtersObj.sorting === el.value) {
+            el.selected = true;
+        }
+    })
+    
+    if (filtersObj.sorting) {
+        let sortValue = filtersObj.sorting;
+        if (sortValue) {
+            createTable(sortingArray (filtered(), sortValue));
+        }
+
+    }
+
+}
+
+function checked () {
+    const clickFilters: NodeListOf<Element> = document.querySelectorAll('.filters-buttons');
+    let checkboxes: NodeListOf<Element> = document.querySelectorAll('.checkbox-input');
+    
+    for (let i= 0; i < checkboxes.length; i++) {
+        checkboxes[i].addEventListener('change', function(event) {
+            
+            if (event.target instanceof HTMLInputElement) {
+                if (event.target.checked) {
+                    if (event.target.form?.className === "checkboxes-country") {                           
+                        countryArray.push(`${event.target.value}`) ;                      
+                        filtersObj.country = countryArray;
+                        filtered();
+                        slider(filtered());
+                    }         
+                    if (event.target.form?.className === "checkboxes-brand") {
+                        brandArray.push(`${event.target.value}`);
+                        filtersObj.brand = brandArray;
+                        filtered();
+                        slider(filtered());
+                        
+                    }
+                    clickFilters.forEach((el) => el.classList.add('active'));
+
+                } else {
+                    if (event.target.form?.className === "checkboxes-country") {
+                        let deleteCountry = `${event.target.value}`;
+                        countryArray = countryArray.filter((name) => name !== deleteCountry);
+                        filtersObj.country = countryArray;
+                        filtered();
+                        slider(filtered());
+                        
+                    }         
+                    if (event.target.form?.className === "checkboxes-brand") {
+                        let deleteBrand = `${event.target.value}`;
+                        brandArray = brandArray.filter((name) => name !== deleteBrand);
+                        filtersObj.brand = brandArray;
+                        filtered();
+                        slider(filtered());
+                        
+                    }
+                }
+            }
+        });
+    }
+}
```

Comment:
You can simplify this code by reducing the duplication and making use of the concept of higher-order functions. Here's a refactored version of your code:
```
function checked() {
  const checkboxes = Array.from(document.querySelectorAll('.checkbox-input'));
  const filtersButtons = Array.from(document.querySelectorAll('.filters-buttons'));

  const handleCheckboxChange = (event) => {
    if (!(event.target instanceof HTMLInputElement)) {
      return;
    }

    const isChecked = event.target.checked;
    const formClass = event.target.form?.className;
    const value = event.target.value;

    if (formClass === "checkboxes-country") {
      filtersObj.country = updateArray(countryArray, value, isChecked);
    } else if (formClass === "checkboxes-brand") {
      filtersObj.brand = updateArray(brandArray, value, isChecked);
    }

    filtered();
    slider(filtered());
    filtersButtons.forEach((el) => {
      if (isChecked) {
        el.classList.add('active');
      } else if (filtersObj.country.length === 0 && filtersObj.brand.length === 0) {
        el.classList.remove('active');
      }
    });
  };

  checkboxes.forEach((checkbox) => {
    checkbox.addEventListener('change', handleCheckboxChange);
  });
}

function updateArray(array, value, shouldAdd) {
  if (shouldAdd) {
    array.push(value);
  } else {
    array = array.filter((name) => name !== value);
  }

  return array;
}
```